libs/remote_storage/src/azure_blob.rs

```diff
@@ -664,3 +675,38 @@ where
         }
     }
 }
+
+/// A Stream wrapper that impls Sync even though the inner doesn't need to.
+struct SyncStream<T: Stream>(std::sync::Mutex<Pin<Box<T>>>);
+
+impl<T: Stream> SyncStream<T> {
+    fn from_pin(inner: Pin<Box<T>>) -> Self {
+        SyncStream(std::sync::Mutex::new(inner))
+    }
+}
+impl<T: Stream + Send> Stream for SyncStream<T> {
+    type Item = T::Item;
+    fn poll_next(
+        self: Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        let mut lock = match self.0.lock() {
+            Ok(lock) => lock,
+            Err(e) => {
+                tracing::info!("Failed to acquire SyncStream lock: {e}");
+                // TODO maybe return an error here instead?
+                return std::task::Poll::Ready(None);
```

Comment:
In general we don't handle poisonings like this, we just unwrap.

Reply:
turned it into an unwrap